drivers: ieee802154_nrf5: fix delayed operation accuracy

lmaciejonczyk opened the pull request:
Align clock accuracy used in CSL calculations to the value in platform.
This is recomended setting for devices working in wide range
temperature. Nevertheless it can be profiled in end product to decrease
CSL window duration and finally the power consumption.

rlubos commented:
CC @jciupis

edmont commented on drivers/ieee802154/ieee802154_nrf5.c:
```diff
@@ -594,7 +594,7 @@ static uint8_t nrf5_get_acc(const struct device *dev)
 	ARG_UNUSED(dev);
 
-	return CONFIG_IEEE802154_DELAY_TRX_ACC;
+	return CONFIG_CLOCK_CONTROL_NRF_ACCURACY;
```
I wonder if in some cases it will be desirable to establish a different value for CSL Accuracy than the main clock accuracy. Users might decide to risk a bit on missing CSL receptions on favor of longer battery life, instead of accounting for worst case drifts.

I would keep `CONFIG_IEEE802154_DELAY_TRX_ACC` option but assign it `CONFIG_CLOCK_CONTROL_NRF_ACCURACY` as default value.

lmaciejonczyk committed:
drivers: ieee802154_nrf5: fix delayed operation accuracy

Align clock accuracy used in CSL calculations to the value in platform.
This is recomended setting for devices working in wide range
temperature. Nevertheless it can be profiled in end product to decrease
CSL window duration and finally the power consumption.

Signed-off-by: Lukasz Maciejonczyk <lukasz.maciejonczyk@nordicsemi.no>

edmont commented on drivers/ieee802154/Kconfig:
```diff
@@ -139,7 +139,7 @@ config IEEE802154_2015
 config IEEE802154_DELAY_TRX_ACC
 	int "Clock accuracy for delayed operations"
-	default 20
+	default CLOCK_CONTROL_NRF_ACCURACY
```
This is not possible since it's an NRF only symbol.

lmaciejonczyk replied:
The option IEEE802154_DELAY_TRX_ACC is used only by nrf driver. It is misplaced. It should be in nrf Kconfig.

edmont replied:
Because no other platform implements CSL yet, but should be possible to configure it.

lmaciejonczyk replied:
The option is used only in the nrf5 implementation of radio api (get_sch_acc specifically). By other platform it can be implemented in any way.

carlescufi commented:
@tbursztyka could you please take a look?

jfischer-no commented on drivers/ieee802154/Kconfig.nrf5:
```diff
@@ -75,4 +75,11 @@ config IEEE802154_NRF5_FCS_IN_LENGTH
 
+config IEEE802154_NRF5_DELAY_TRX_ACC
+	int "Clock accuracy for delayed operations"
+	default CLOCK_CONTROL_NRF_ACCURACY
```
That can be default to 500 now
```
config CLOCK_CONTROL_NRF_ACCURACY                                                                                                            
        int 
        default 500 if CLOCK_CONTROL_NRF_K32SRC_500PPM                                                                                       
        default 250 if CLOCK_CONTROL_NRF_K32SRC_250PPM
....
```

jfischer-no commented on drivers/ieee802154/ieee802154_nrf5.c:
```diff
@@ -594,7 +594,7 @@ static uint8_t nrf5_get_acc(const struct device *dev)
 	ARG_UNUSED(dev);
 
-	return CONFIG_IEEE802154_DELAY_TRX_ACC;
+	return CONFIG_IEEE802154_NRF5_DELAY_TRX_ACC;
```
This produces warning and breaks CI, since it can be larger than UINT8_MAX.
https://github.com/zephyrproject-rtos/zephyr/pull/44402/checks?check_run_id=5769581818
